src/System.Private.CoreLib/src/System/Diagnostics/Eventing/EventPipeController.cs

```diff
@@ -0,0 +1,173 @@
+    ///  - COMPlus_EventPipeConfig : Provides the configuration in xperf string form for which providers/keywords/levels to be enabled.
+    ///                              If not specified, the default configuration is used.
+    ///  - COMPlus_EventPipeOutputFile : The full path to the netperf file to be written.
+    /// Once the configuration is set and this controller is enabled, tracing is enabled by creating a marker file that this controller listens for.
+    /// Tracing is disabled by deleting the marker file.  The marker file is the target trace file path with ".ctl" appended to it.  For example,
+    /// if the trace file is /path/to/trace.netperf then the marker file is /path/to/trace.netperf.ctl.
+    /// This listener does not poll very often, and thus takes time to enable and disable tracing.  This is by design to ensure that the listener does
+    /// not starve other threads on the system.
+    /// NOTE: If COMPlus_EnableEventPipe != 4 then this listener is not created and does not add any overhead to the process.
+    /// </summary>
+    internal sealed class EventPipeController
+    {
+        private const string MarkerFileExtension = ".ctl";
+        private const int PollingIntervalMilliseconds = 10000; // 10 seconds
+        private const uint DefaultCircularBufferMB = 1024; // 1 GB
+        private readonly char[] ProviderConfigDelimiter = new char[] { ',' };
+        private readonly char[] ConfigComponentDelimiter = new char[] { ':' };
+
+        private readonly EventPipeProviderConfiguration[] DefaultProviderConfiguration = new EventPipeProviderConfiguration[]
+        {
+            new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntime", 0x4c14fccbd, 5),
+            new EventPipeProviderConfiguration("Microsoft-Windows-DotNETRuntimePrivate", 0x4002000b, 5),
+            new EventPipeProviderConfiguration("Microsoft-DotNETCore-SampleProfiler", 0x0, 5)
+        };
+
+        private static EventPipeController s_controllerInstance;
+
+        private string m_traceFilePath;
+        private string m_markerFilePath;
+        private bool m_markerFileExists;
+
+        internal static void Initialize()
+        {
+            // Don't allow failures to propagate upstream.
+            // Instead, ensure program correctness without tracing.
+            try
+            {
+                if (s_controllerInstance == null)
+                {
+                    string strEnabledValue = CompatibilitySwitch.GetValueInternal("EnableEventPipe");
+                    if (strEnabledValue != null)
+                    {
+                        int enabledValue = Convert.ToInt32(strEnabledValue);
+                        if (enabledValue == 4)
+                        {
+                            s_controllerInstance = new EventPipeController();
+                        }
+                    }
+                }
+            }
+            catch { }
+        }
```

Comment:
I would simplify the method and reduce nesting to:

```cs
internal static void Initialize()
{
    if (s_controllerInstance != null)
        return;

    try
    {
        if(int.TryParse(CompatibilitySwitch.GetValueInternal("EnableEventPipe"), out int flag) && flag == Constants.EventPipeInitiallyDisabled)
            s_controllerInstance = new EventPipeController();
    }
    catch {}    
}
```

Also I would check if `CompatibilitySwitch.GetValueInternal` can throw. If not, I would remove the catch block.